main.tf

```diff
@@ -12,6 +13,25 @@ terraform {
   backend "inmem" {}
 }
 
+provider "imagetest" {
+  harnesses = {
+    k3s = {
+      networks = {
+        // wire in k3d's default network where the registry lives
+        "k3d-default" = { name = "k3d-k3s-default" }
+      }
+      registries = {
+        # Mirror the var.target_repository host registry to the local registry.
+        # This ensures the images that are pushed from the host registry are
+        # mirrored to the internal hostname:port registry.
+        "${element(split("/", var.target_repository), 0)}" = {
+          mirror = { endpoints = ["http://${element(split(":", var.target_repository), 0)}:5000"] }
```

Comment:
Does this work when the sandbox network has not been connected to the existing registry network?

Reply:
good catch! 

turns out when you spin up the registry with the same cluster create command, it attaches the registry to the cluster network (`k3s-k3d-default`).

since CI also spins up the registry with the same `host:port` combo, the only thing we need here is the network attachment